### PR TITLE
test(acceptance): add annotations with nested attributes into scenario for CRD annotations

### PR DESCRIPTION
**Pull request by pmacik**
### Motivation

#888 Suggests some scenarios to be covered in acceptance tests. The nested arguments scenario as suggested is not covered by any current acceptance test.

### Changes

This PR:
* Enhances one of the scenarios with binding annotations for nested arguments as suggested in #888

### Testing

PR checks should do

### CODE REVIEW

File: test/acceptance/features/bindAppToServiceAnnotations.feature [hunk 1]
```diff
@@ -518,8 +518,15 @@ Feature: Bind an application to a service using annotations
                 name: binding-app-via-gvk-service
                 annotations:
                     service.binding/host_internal_db: path={.spec.host_internal_db}
+                    service.binding/environment: path={.spec.userLabels.environment}
+                    service.binding/dbCredentials: path={.status.data.dbCredentials}
             spec:
                 host_internal_db: internal.db.stable.example.com
+                userLabels:
```
Comment (pedjak) [inline test/acceptance/features/bindAppToServiceAnnotations.feature:525]: this scenario has a focus on something fully different, and hence I would not add broader its focus. Instead, let's add a separate test or find another which focus is on exposing binding data using annotations.

**Review by pedjak — CHANGES_REQUESTED** [review]
please rephrase commit message to state exactly what kind of test was added and please do not refer GitHub issue in it.

File: test/acceptance/features/bindAppToServiceAnnotations.feature (hunk 1 above)
Comment (pmacik) [reply test/acceptance/features/bindAppToServiceAnnotations.feature]: Good point, thx @pedjak 
Updated a different scenario

**Review by pedjak — APPROVED** [review]
/lgtm
/approve

**Pull request merged**